lib/internal/cluster/shared_handle.js

```diff
@@ -1,46 +1,49 @@
 'use strict';
+const { Map } = primordials;
 const assert = require('internal/assert');
 const dgram = require('internal/dgram');
 const net = require('net');
 
-module.exports = SharedHandle;
-
-function SharedHandle(key, address, port, addressType, fd, flags) {
-  this.key = key;
-  this.workers = [];
-  this.handle = null;
-  this.errno = 0;
-
-  let rval;
-  if (addressType === 'udp4' || addressType === 'udp6')
-    rval = dgram._createSocketHandle(address, port, addressType, fd, flags);
-  else
-    rval = net._createServerHandle(address, port, addressType, fd, flags);
-
-  if (typeof rval === 'number')
-    this.errno = rval;
-  else
-    this.handle = rval;
+class SharedHandle {
+  constructor(key, address, { port, addressType, fd, flags }) {
+    this.key = key;
+    this.workers = new Map();
```

Comment:
Wouldn't a `WeakSet` (or `SafeWeakSet`) be a better fit than `Map` here?

Reply:
Since in map we are relying on a key, that is in our case is `id`. On if deep-clone an object itself and pass it then also the lookup will not be evaluated to true. Ref:
```
> a = {b: 'some key'};
{ b: 'some key' }
> st = new WeakSet()
WeakSet { <items unknown> }
> st.add(a)
WeakSet { <items unknown> }
> st
WeakSet { <items unknown> }
> m = Object.assign({}, a)
{ b: 'some key' }
> st.has(m)
false
> st.has(a)
true
>
```